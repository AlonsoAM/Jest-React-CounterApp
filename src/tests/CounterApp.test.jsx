import React from "react";
import { shallow } from "enzyme";
import CounterApp from "../CounterApp";

describe("Pruebas en el <CounterApp />", () => {
    let wrapper = shallow(<CounterApp />);
    beforeEach(() => {
        wrapper = shallow(<CounterApp />);
    });

    test("Debe mostar <CounterApp /> correctmante", () => {
        expect(wrapper).toMatchSnapshot();
    });

    test("Debe mostrar el valor por defecto de 100 ", () => {
        const wrapper = shallow(<CounterApp value={100} />);
        const conuterText = wrapper.find("h2").text().trim();
        console.log(`xxxx${conuterText}xxxx`);
        expect(conuterText).toBe("100");
    });

    test("Debe incrementar con el botón +1", () => {
        wrapper.find("button").at(0).simulate("click");
        // console.log(btn1.html());
        const conuterText = wrapper.find("h2").text().trim();
        expect(conuterText).toBe("11");
    });

    test("Debe decrementar con el botón -1", () => {
        wrapper.find("button").at(2).simulate("click");
        const conuterText = wrapper.find("h2").text().trim();
        expect(conuterText).toBe("9");
    });
});
