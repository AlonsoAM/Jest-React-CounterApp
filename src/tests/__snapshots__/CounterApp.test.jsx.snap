// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <CounterApp /> Debe mostar <CounterApp /> correctmante 1`] = `
<div>
  <h1>
     Counter App 
  </h1>
  <h2>
     
    10
     
  </h2>
  <button
    onClick={[Function]}
  >
     +1 
  </button>
  <button
    onClick={[Function]}
  >
     Reset 
  </button>
   
  <button
    onClick={[Function]}
  >
     -1 
  </button>
</div>
`;
